### PR TITLE
Measure Report documentation

### PR DESCRIPTION
**Pull request by maimoonak**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #[issue number] 

**Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: MEASURE.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
+## MEASURE
+The [Measure](http://hl7.org/fhir/R4/measure.html) is a FHIR resource which represents definition- for calculation of an indicator. The Measure uses a logic Library (CQL) that contains the calculation logic for measure.
+
+A basic measure report can comprise of following components
```
Comment (f-odhiambo) [inline MEASURE.md:4]: I think let's use real examples for the purposes of this illustration

File: MEASURE.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+**scoring**: proportion | ratio | continuous-variable | cohort
+
+**group**: The section of report defining the stats for a specific indicator (may be disaggregated by stratifier). Each group has following components
+- id: Group name/id.
+- population: The calculations (or name of variable in CQL that defines the value) for each of population components i.e. initial-population, denominator, numerator
+- stratifier: The disaggregations for given population numerator. i.e. by age, by month, by education etc 
+- supplementalData: Any extra data or intermediate calculation to be output to final report
+
+
+## MEASURE LIBRARY
```
Comment (f-odhiambo) [inline MEASURE.md:102]: I think this Title should read CQL Expression containing decision logic or Measure to compute

File: MEASURE.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+A [elm REST app](https://github.com/cqframework/cql-translation-service/blob/master/README.md) that can be used to run elm microservice and convert CQL via a REST API. 
+
+
+**CQL to ELM JAVA Translator**:
+A [elm java app](https://github.com/cqframework/clinical_quality_language/blob/master/Src/java/READM.md) that can be used to elm translator on files and get an output. Instructions can be found [here](https://github.com/cqframework/cql-execution#to-execute-your-cql)
+
+**Note**: Above approaches output a json elm which then need to be base64 decoded and copied to the [Library](http://hl7.org/fhir/R4/library.html) content as Attachment.
+
+**Fhir-Resource on FhirCore**:
+The [fhir-resources](https://github.com/opensrp/fhir-resources/fhircore-testing/src/test/resources/measure-report/household-members.feature) repository has a testing module which allows you to not only get the complete Library resource to directly save to server but also allows to test the Measure output and make changes on the fly. Check the cucumber tests fhircore-testing/src/test/resources/measure-report/household-members.feature.
```
Comment (f-odhiambo) [inline MEASURE.md:143]: Can we add a link to the line number or actual file containing the function to where this code is being tested

File: MEASURE.md [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+A [elm java app](https://github.com/cqframework/clinical_quality_language/blob/master/Src/java/READM.md) that can be used to elm translator on files and get an output. Instructions can be found [here](https://github.com/cqframework/cql-execution#to-execute-your-cql)
+
+**Note**: Above approaches output a json elm which then need to be base64 decoded and copied to the [Library](http://hl7.org/fhir/R4/library.html) content as Attachment.
+
+**Fhir-Resource on FhirCore**:
+The [fhir-resources](https://github.com/opensrp/fhir-resources/fhircore-testing/src/test/resources/measure-report/household-members.feature) repository has a testing module which allows you to not only get the complete Library resource to directly save to server but also allows to test the Measure output and make changes on the fly. Check the cucumber tests fhircore-testing/src/test/resources/measure-report/household-members.feature. 
+
+**FhirCore Unit Tests**
+The CQL can also be translated to [Library] using an approach like fhircore as in unit tests org.smartregister.fhircore.quest.CqlContentTest#runCqlLibraryTestForPqMedication. A complete Library resource is output to console as a result.
```
Comment (f-odhiambo) [inline MEASURE.md:146]: Same comments as above

File: MEASURE.md [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+**measure**: The Measure.url for which this report was generated i.e. http://fhir.org/guides/who/anc-cds/Measure/HOUSEHOLDIND01
+
+**period**: Measure period which was sent for date filter i.e. reporting period start and end. The Measure interval has closed boundaries. Read details [here](https://cql.hl7.org/02-authorsguide.html#interval-values) 
+
+**group**: The calculated value for each Measure indicator with
+ - id: id/name of group/indicator/stratifier as defined in Meaure. 
+ - count: Calculated value from CQL for given variable
+ - measureScore: The percent/ratio of calculated value i.e. numerator/denomintor. Note that for stratifier the score denominator is stratifier denomintor rather than group denominator
+ - 
+**stratifier**: Count for given indicator disaggregated by each type. The stratifier misses the values where counts are zero. Hence if stratifier has predefined criteria, each should be a calculated as separate group.
```
Comment (f-odhiambo) [inline MEASURE.md:236]: Good start. I think we can follow perhaps the following flow 
1. Measure  - What a measure and what is it used for 
2. CQL - How we author a CQL script, where to test 
3. How to load CQL Library to FHIR Core  - convert to ELM and load on HAPI, update configs 
4. Invoking $evalute_measure  - How do we call this API on FHIR Core 
5. MeasureReport  - What are the components of interest from a generated MeasureReport
6. Reporting interface on FHIR Core  - How does the reporting interface consume MeasureReport

File: MEASURE.md [hunk 6]
```diff
@@ -0,0 +1,236 @@
+## MEASURE
```
Comment (f-odhiambo) [inline MEASURE.md:1]: Let's use sentence case i.e.  Measure

File: MEASURE.md [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+
+**Note**: Above approaches output a json elm which then need to be base64 decoded and copied to the [Library](http://hl7.org/fhir/R4/library.html) content as Attachment.
+
+**Fhir-Resource on FhirCore**:
+The [fhir-resources](https://github.com/opensrp/fhir-resources/fhircore-testing/src/test/resources/measure-report/household-members.feature) repository has a testing module which allows you to not only get the complete Library resource to directly save to server but also allows to test the Measure output and make changes on the fly. Check the cucumber tests fhircore-testing/src/test/resources/measure-report/household-members.feature. 
+
+**FhirCore Unit Tests**
+The CQL can also be translated to [Library] using an approach like fhircore as in unit tests org.smartregister.fhircore.quest.CqlContentTest#runCqlLibraryTestForPqMedication. A complete Library resource is output to console as a result.
+
+## MEASURE REPORT
```
Comment (f-odhiambo) [inline MEASURE.md:148]: Let's use sentence case i.e.  MeasureReport

**Review by f-odhiambo — COMMENTED** [review]
Check out my comments

File: MEASURE.md [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+**group**: The calculated value for each Measure indicator with
+ - id: id/name of group/indicator/stratifier as defined in Meaure. 
+ - count: Calculated value from CQL for given variable
+ - measureScore: The percent/ratio of calculated value i.e. numerator/denomintor. Note that for stratifier the score denominator is stratifier denomintor rather than group denominator
+ - 
+**stratifier**: Count for given indicator disaggregated by each type. The stratifier misses the values where counts are zero. Hence if stratifier has predefined criteria, each should be a calculated as separate group. 
+
+## Screenshots
+
+<img width="300" height="500" src="https://user-images.githubusercontent.com/4829880/188478590-93474727-4ef2-4ba7-acfc-8eb6ac9e32d5.png"/> <img width="300" height="500" src="https://user-images.githubusercontent.com/4829880/188478602-e3941d73-4582-43a2-bc24-423eb1253fe0.png"/>
```
Comment (f-odhiambo) [inline MEASURE.md:232]: Let's make these images width="200" height="400" to make them more legible

File: MEASURE.md [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+
+**Fhir-Resource on FhirCore**:
+The [fhir-resources](https://github.com/opensrp/fhir-resources/blob/main/fhircore-testing) repository has a testing module which allows to not only get the complete Library resource to directly save to server but also allows to test the Measure output and make changes on the fly. Check the cucumber tests [Feature File](https://github.com/opensrp/fhir-resources/blob/main/fhircore-testing/src/test/resources/measure-report/household-members.feature), the [Test Code File](https://github.com/opensrp/fhir-resources/blob/main/fhircore-testing/src/test/kotlin/com/fhircore/resources/testing/measure/HouseholdMembersMeasureTest.kt#L28) and the [Convertor Util Method](https://github.com/opensrp/fhir-resources/blob/main/fhircore-testing/src/test/kotlin/com/fhircore/resources/testing/CqlUtils.kt#L31)
+
+**FhirCore Unit Tests**
+The CQL can also be translated to Library using an approach as used by fhircore as in [CQL Content Tests](https://github.com/opensrp/fhircore/blob/main/android/quest/src/test/java/org/smartregister/fhircore/quest/CqlContentTest.kt#L57). A complete Library resource is output to console as a result.
+
+## Measure Report
```
Comment (f-odhiambo) [inline MEASURE.md:139]: MeasureReport is one word

**Review by pld — APPROVED** [review]

**Review by ellykits — APPROVED** [review]

**Pull request merged**